logic/and.js

```diff
@@ -0,0 +1,26 @@
+/** @license ISC License (c) copyright 2017 original and current authors */
+/** @author Ian Hofmann-Hicks (evil) */
+
+const isFunction = require('../predicates/isFunction')
+const isSameType = require('../predicates/isSameType')
+
+const Pred = require('../crocks/Pred')
+
+const curry = require('../helpers/curry')
+const identity = require('../combinators/identity')
+
+const ifElse = require('./ifElse')
+
+const predOrFunc =
+  ifElse(isFunction, identity, x => x.runWith)
+
+// and : (a -> Boolean) | Pred -> (a -> Boolean) | Pred -> a -> Boolean
+function and(f, g) {
+  if(!((isFunction(f) || isSameType(Pred, f)) && (isFunction(g) || isSameType(Pred, g)))) {
+    throw new TypeError('and: Preds or predicate functions required for first two arguments')
```

Comment:
I think I am goin to leave this as both types are accepted.
And this lets you do things like:
[this](https://gist.github.com/evilsoft/0f720f43af2000d2c480a712f7db4ebd)